explainer-client/src/main/scala/ExplainEditorJS.scala

```diff
@@ -45,74 +45,112 @@ object ExplainEditorJS {
 
     val explainer = Var(Explainer)
 
-    def init(id: String): Future[Explainer] = {
+    def extractExplainer(id: String): Future[Explainer] = {
       Ajaxer[ExplainerApi].load(id).call()
     }
 
-    def saveContent(id: String, explainer: ExplainerUpdate) = {
+    def updateFieldContent(id: String, explainer: ExplainerUpdate) = {
       Ajaxer[ExplainerApi].update(id, explainer.field, explainer.value).call()
     }
 
-  }
-
-  val headline: TypedTag[Input] = input(
-    id:="new-atom",
-    placeholder:="headline",
-    autofocus:=true
-  )
-
-  val body: TypedTag[TextArea] = textarea(
-    id:="new-atom",
-    rows:=18,
-    cols:=50,
-    maxlength:=1800,
-    placeholder:="body",
-    autofocus:=true
-  )
-
-  def templateHeader(explainerId: String, explainer: Explainer) = {
+    def createNewExplainer() = {
+      Ajaxer[ExplainerApi].create().call()
+    }
 
-    def turnOnPresenceFor(area: String, field: Element) = {
-      field.onfocus = (x: FocusEvent) => {
-        presenceClient.enter("explain-" + explainerId, area)
-      }
+    def publish(id: String): Future[Explainer] = {
+      Ajaxer[ExplainerApi].publish(id).call()
+    }
 
-      val indicatorId = s"presence-indicator-$area"
-      val fieldPresenceIndicator = div(`class` := "field-presence-indicator", id := indicatorId)()
+  }
 
-      presenceClient.on("visitor-list-updated", { data: js.Object =>
-        val stateChange = upickle.default.read[StateChange](js.JSON.stringify(data))
+  def presenceEnvelop(explainerId: String ,area: String, field: Element) = {

```

Comment:
I'm not sure this is the most understandable name. Is this the function that adds the presence indicator? Maybe the old function name could work better. But I'm sure you had a good reason for changing it!